src/Features/AssignableAsset.php

```diff
@@ -87,20 +90,49 @@ public function canUpdateItem()
     public static function getAssignableVisiblityCriteria()
```

Comment:
> as it assumes everything from it is WHERE.

As far as I see `getDefaultWhereCriteria()` always return somthing that corresponds to a `WHERE` clause contents. There are even some subqueries to handle lack of JOIN support here, for instance with the usage of `buildParentCondition()` methods.

So, unless I miss something, it would be better to use a subquery here, in order to be able to use the `WHERE` part without having to handle any `JOIN`.